### PR TITLE
Stop using vfork with musl libc and credentials

### PR DESCRIPTION
**Pull request by stephentoub**
Port https://github.com/dotnet/corefx/pull/41457 to master.  We should have started with this master; we can later refine it if desired.
#41432

### CODE REVIEW

**Comment by joshudson** [comment]
Well I'll tell you. I expected these might be different because I was in some discussion with the musl maintainers about an extension to posix_spawnp.

**Comment by stephentoub** [comment]
As noted we can change it later if appropriate. But in the meantime we should not have fixes in release/3.0 that aren't in master.

File: src/Native/Unix/System.Native/pal_process.c [hunk 1]
```diff
@@ -339,7 +339,21 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
...
     if ((processId = vfork()) == 0) // processId == 0 if this is child process
+#else
+    // musl libc has an undocumented failure mode around setuid(); we must exclude it.
+    if (setCredentials)
+    {
+        processId = fork();
+    }
+    else
+    {
+        processId = vfork();
```
Comment (krwq) [inline src/Native/Unix/System.Native/pal_process.c]: this could be `processId = setCredentials ? fork() : vfork()`

Comment (joshudson) [reply src/Native/Unix/System.Native/pal_process.c]: That's exactly what the comment I was asked to remove said to not do. I don't trust it.

**Pull request merged**